project.mml

```diff
@@ -2278,6 +2277,30 @@ Layer:
     properties:
       cache-features: true
       minzoom: 14
+# Render railway crossings except if they concern tramways only
+  - id: railway-crossings
+    geometry: point
+    <<: *extents
+    Datasource:
+      <<: *osm2pgsql
+      table: &railway-crossing_sql |-
+        (SELECT DISTINCT ON (crossing.way)
+            'railway_' || crossing.railway AS feature,
+            crossing.way AS way,
+            crossing.access AS access,
+            crossing.railway AS type,
+            crossing.railway AS railway,
+            track.railway AS int_lc_type
+          FROM planet_osm_point crossing
+            JOIN planet_osm_line track
+              ON ST_DWithin(crossing.way, track.way, 0.1) -- Assumes Mercator
+          WHERE crossing.railway IN ('crossing', 'level_crossing')
+            AND track.railway IS NOT NULL AND track.railway <> 'tram'
+          ORDER BY crossing.way
```

Comment:
Although this is not a newly introduced problem i would like to mention that we generally want to use explicit ordering in queries where the order of features returned matters (which is the case here because the symbols are blocking each other).  It would therefore be advisable to order also by (a) priority based on railway type and (b) osm_id to always have a defined drawing order.